### PR TITLE
feature/#13/bfs_algorithm

### PR DESCRIPTION
**Pull request by Segmentree**
# Description

Add `bfs` algorithm to the library

## List of changes :

- [x] Add `bfs` method to graph algorithms
- [x] Add unit test to assert all vertices visited

### CODE REVIEW

File: tests/graphs/bfs.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    //    2     3
+    //   /     / \
+    //  0     4   1
+    let mut vec = vec![];
+    vec.push(vec![2]);
+    vec.push(vec![3]);
+    vec.push(vec![0, 5]);
+    vec.push(vec![1, 4, 5]);
+    vec.push(vec![3]);
+    vec.push(vec![2, 3]);
```
Comment (rbruno95) [inline tests/graphs/bfs.rs]: @Segmentree 

```rs
  let mut vec = vec![
      vec![2],
      vec![3],
      vec![0, 5],
      vec![1, 4, 5],
      vec![3],
      vec![2, 3]
  ];
```

File: src/algorithms/graphs/bfs.rs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+use std::collections::VecDeque;
+
+#[allow(dead_code)]
+pub fn bfs(adj: &mut Vec<Vec<usize>>, vis: &mut Vec<bool>, v: usize) {
```
Comment (rbruno95) [inline src/algorithms/graphs/bfs.rs]: @Segmentree 
the initial node should be called `s` from the source, don't you think?
adj shouldn't be mutable

File: src/algorithms/graphs/bfs.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    vis[s] = true;
+    while !q.is_empty() {
+        let u = q.pop_front().unwrap();
+        vis[u] = true;
+        for i in 0..adj[u].len() {
+            let v = adj[u][i];
+            if !vis[v] {
+                q.push_back(v);
+            }
+        }
```
Comment (rbruno95) [inline src/algorithms/graphs/bfs.rs]: @Segmentree could you change this for as in the dfs?

Comment (Segmentree) [reply src/algorithms/graphs/bfs.rs]: @rbruno95 I think the correct way to say that is `for how in the dfs` but yes no problem

Comment (rbruno95) [reply src/algorithms/graphs/bfs.rs]: @Segmentree I think I wrote it correctly, I've just checked it using DeepL

**Review by rbruno95 — APPROVED** [review]

**Pull request merged**